adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -726,6 +726,93 @@ public void onError(Exception exc) {
         return authenticationResult.get();
     }
 
+    /**
+     * This is sync function. It will first look at the cache and automatically
+     * checks for the token expiration. Additionally, if no suitable access
+     * token is found in the cache, but refresh token is available, the function
+     * will use the refresh token automatically. This method will not show UI
+     * for the user. If prompt is needed, the method will return an exception
+     *
+     * @param resource required resource identifier.
+     * @param clientId required client identifier.
+     * @param userId   UserID obtained from
+     *                 {@link AuthenticationResult #getUserInfo()}
+     * @param forceRefresh when true, access token is renewed using broker if available; otherwise, uses local refresh token
+     * @return A {@link Future} object representing the
+     * {@link AuthenticationResult} of the call. It contains Access
+     * Token,the Access Token's expiration time, Refresh token, and
+     * {@link UserInfo}.
+     * @throws AuthenticationException If silent request fails to get the token back.
+     * @throws InterruptedException    If the main thread is interrupted before or during the activity.
+     */
+    public AuthenticationResult acquireTokenSilentSync(String resource, String clientId, String userId, boolean forceRefresh)
```

Comment:
Can the alternative implementations of `acquireTokenSilentSync` call this (passing `false`) to achieve their current behavior? If yes, can we turn those implementations into thin wraps around this method? (or possibly an internal call which allows for the `apiEvent` to be passed?)

Reply:
Yep.